lib/puppet/provider/grafana_organization/grafana.rb

```diff
@@ -0,0 +1,119 @@
+#    Copyright 2015 Mirantis, Inc.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+#
+require 'json'
+
+require File.expand_path(File.join(File.dirname(__FILE__), '..', 'grafana'))
+
+Puppet::Type.type(:grafana_organization).provide(:grafana, parent: Puppet::Provider::Grafana) do
+  desc 'Support for Grafana organizations'
+
+  defaultfor kernel: 'Linux'
+
+  def organizations
```

Comment:
Ditto for datasources. When you define a method like this (especially using a local variable with the same name later) it's very confusing when this object isn't actually a model. organizations and organization should be puppet getters/setters for the resource, and fetch_organizations etc should be a 'retrieval' from an external authority

Reply:
Datasources was there from the original code on the current master branch of vox. If we're planning on pushing this up as a PR to voxpopuli, we probably shouldn't edit their code, just add our changes in.